src/utils/daos.c

```diff
@@ -0,0 +1,870 @@
+	 * op == CONT_LIST_ATTRS
+	 * op == CONT_DEL_ATTR
+	 * op == CONT_GET_ATTR
+	 * op == CONT_SET_ATTR
+	 * op == CONT_CREATE_SNAP
+	 * op == CONT_LIST_SNAPS
+	 * op == CONT_DESTROY_SNAP
+	 * op == CONT_ROLLBACK
+	 */
+
+bad_cont_destroy:
+bad_cont_close:
+bad_cont_open:
+bad_cont_create:
+	/* Pool disconnect  in normal and error flows: preserve rc */
+	rc2 = daos_pool_disconnect(pool, NULL);
+	if (rc2 != 0) {
+		fprintf(stderr, "Pool disconnect failed : %d\n", rc2);
+	}
+	if (rc == 0)
+		rc = rc2;
+bad_pool_connect:
+bad_opt_free_mdsrv:
+	daos_rank_list_free(ap->mdsrv);
+	return rc;
+}
+
+static int
+help_hdlr(struct cmd_args_s *ap)
+{
+	FILE *stream;
+
+	assert(ap != NULL);
+
+	stream = (ap->ostream != NULL) ? ap->ostream : stdout;
+
+	fprintf(stream, "\
+usage: daos RESOURCE COMMAND [OPTIONS]\n\
+resources:\n\
+	  pool             pool\n\
+	  container (cont) container\n\
+	  help             print this message and exit\n");
+
+	fprintf(stream, "\n\
+pool commands:\n\
+	  list-containers  list all containers in pool\n\
+	  list-cont\n\
+	  query            query a pool\n\
+	  stat             get pool statistics\n\
+	  list-attrs       list pool user-defined attributes\n\
+	  get-attr         get pool user-defined attribute\n");
+
+	fprintf(stream, "\
+pool options:\n\
+	--pool=UUID        pool UUID \n\
+	--group=STR        pool server process group (\"%s\")\n\
+	--svc=RANKS        pool service replicas like 1,2,3\n\
+	--attr=NAME        pool attribute name to get\n",
+	default_group);
+
+	fprintf(stream, "\n\
+container (cont) commands:\n\
+	  create           create a container\n\
+	  destroy          destroy a container\n\
+	  list-objects     list all objects in container\n\
+	  list-obj\n\
+	  query            query a container\n\
+	  stat             get container statistics\n\
+	  list-attrs       list container user-defined attributes\n\
+	  del-attr         delete container user-defined attribute\n\
+	  get-attr         get container user-defined attribute\n\
+	  set-attr         set container user-defined attribute\n\
+	  create-snap      create container snapshot (optional name)\n\
+			   at most recent committed epoch\n\
+	  list-snaps       list container snapshots taken\n\
+	  destroy-snap     destroy container snapshots\n\
+			   by name, epoch or range\n\
+	  rollback         roll back container to specified snapshot\n");
+
+	fprintf(stream, "\
+container (cont) options:\n\
+	  <pool options>   (--pool, --group, --svc)\n\
+	--cont=UUID        container UUID\n\
+	--attr=NAME        container attribute name to set, get, del\n\
+	--value=VALUESTR   container attribute value to set\n\
+	--snap=NAME        container snapshot (create/destroy-snap, rollback)\n\
+	--epc=EPOCHNUM     container epoch (destroy-snap, rollback)\n\
+	--eprange=B-E      container epoch range (destroy-snap)\n");
+	return 0;
+}
+
+int
+main(int argc, char *argv[])
+{
+	int			rc = 0;
+	command_hdlr_t		hdlr = NULL;
+	struct cmd_args_s	dargs;
```

Comment:
dargs = {0};
then you don't need the init function?

Reply:
need to initialize a few fields to something other than 0 or NULL. Will remove the init function and just put those small number of lines in common_op_parse_hdlr()